CPAC/resources/configs/pipeline_config_abcd-options.yml

```diff
@@ -0,0 +1,371 @@
+        fnirt_config: T1_2_MNI152_2mm
+
+        # Reference mask with 2mm resolution to be used during FNIRT-based brain extraction in ABCD-options pipeline.
+        ref_mask_res-2: /ABCD_template/MNI152_T1_2mm_brain_mask_dil.nii.gz
+
+        # Template with 2mm resolution to be used during FNIRT-based brain extraction in ABCD-options pipeline.
+        T1w_template_res-2: /ABCD_template/MNI152_T1_2mm.nii.gz
+
+    applywarp: 
+
+      # using: ANTS or FSL. Choose the applywarp tool regardless of which tool to calculate transforms
+      using: FSL
+
+  functional_registration: 
+
+    coregistration: 
+
+      # reference: 'brain' or 'restore-brain'
+      # In ABCD-options pipeline, 'restore-brain' is used as coregistration reference
+      reference: restore-brain
+
+      # interpolation for coregistration
+      interpolation: spline
+
+      # degree of freedom for coregistration
+      dof: 12
+
+      func_input_prep: 
+
+        # Choose whether to use functional brain or skull as the input to functional-to-anatomical registration
+        reg_with_skull: On
+
+        # Choose whether to use the mean of the functional/EPI as the input to functional-to-anatomical registration or one of the volumes from the functional 4D timeseries that you choose.
+        # input: ['Mean Functional', 'Selected_Functional_Volume']
+        input: [Selected_Functional_Volume]
+
+      boundary_based_registration: 
+
+        # this is a fork point
+        #   run: [On, Off] - this will run both and fork the pipeline
+        run: [Off]
+
+    func_registration_to_template: 
+
+      output_resolution: 
+
+        # The resolution (in mm) to which the preprocessed, registered functional timeseries outputs are written into.
+        # NOTE:
+        #   selecting a 1 mm or 2 mm resolution might substantially increase your RAM needs- these resolutions should be selected with caution.
+        #   for most cases, 3 mm or 4 mm resolutions are suggested.
+        # NOTE:
+        #   this also includes the single-volume 3D preprocessed functional data,
+        #   such as the mean functional (mean EPI) in template space
+        func_preproc_outputs: 2mm
+
+        # The resolution (in mm) to which the registered derivative outputs are written into.
+        # NOTE:
+        #   this is for the single-volume functional-space outputs (i.e. derivatives)
+        #   thus, a higher resolution may not result in a large increase in RAM needs as above
+        func_derivative_outputs: 2mm
+
+      target_template: 
+
+        # using: ['T1_template', 'EPI_template', 'ABCD']
+        # this is a fork point
+        # NOTE:
+        #   this will determine which registration transform to use to warp the functional
+        #   outputs and derivatives to template space.
+        #   if using 'ABCD', apply motion correction, func-to-anat, anat-to-template transforms 
+        #   on each of raw functional volume same as ABCD-options pipeline.
+        using: [ABCD]
```

Comment:
@sgiavasis Do you think this name is appropriate? Do we need to give this apply xfm method another name w/o specifying "ABCD"?

Reply:
I think it may be better to name it something more descriptive re: the method itself, but of course clearly cite DCAN + ABCD in the config, code, and user docs.

But before we settle on the name- I wonder if this option would be better off in a different nest layer but still under functional registration to template. Because `target_template` is about the reference used in the apply warp, not the transform itself.

Maybe between `output_resolution` and `target_template`, there can be a level called `transformation` - and the default would be the coreg and T1-to-template combined as one, and the "ABCD" one would just be the addition of motion correction.

What do you think?

Reply:
I agree that putting this option in another config level will make it clearer. The reason I set up in this way is that the function [warp_timeseries_to_T1template_abcd](https://github.com/FCP-INDI/C-PAC/blob/feature/abcd_v1.8/CPAC/registration/registration.py#L2543) is in the same level as [warp_timeseries_to_T1template](https://github.com/FCP-INDI/C-PAC/blob/feature/abcd_v1.8/CPAC/registration/registration.py#L2480) and `warp_timeseries_to_T1template` depends on the config field `[func_registration_to_template][target_template][using]`, so I add this option in the same config field. If we specify "ABCD", it will only run `warp_timeseries_to_T1template_abcd` but not `warp_timeseries_to_T1template`. If we are going to have another layer, how can we change the docstring so that the regular `warp_timeseries_to_T1template` won't run in ABCD-options pipeline?

Reply:
Resolved in https://github.com/FCP-INDI/C-PAC/pull/1500